### PR TITLE
Introduce Contrast Adaptive Sharpening post process effect

### PR DESCRIPTION
**Pull request by RyeMutt**
This introduces Contrast Adaptive Sharpening from https://github.com/GPUOpen-Effects/FidelityFX-CAS

Setting range is 0.0 - 1.0 with 0.0 disabling the effect/shader pass

0.0:
Snapshot_004
0.4:
Snapshot_002
1.0:
Snapshot_003

### CODE REVIEW

**Review by Geenz — APPROVED** [review]
There's definitely some eyebrow raising stuff in LLPipeline, but it appears that this is how AMD wants you to integrate shaders like this.  So it's fine, and will probably take more time to separate the AMD specific methods from their GLSL than it's probably worth.

Tested this locally - seems to work as I'd expect.

File: indra/newview/pipeline.cpp [hunk 1]
```diff
@@ -7133,6 +7147,50 @@ void LLPipeline::generateGlow(LLRenderTarget* src)
...
+    // Bind setup:
+    dst->bindTarget();
+
+    sharpen_shader->bind();
+
+    {
+        static LLStaticHashedString cas_param_0("cas_param_0");
+        static LLStaticHashedString cas_param_1("cas_param_1");
+        static LLStaticHashedString out_screen_res("out_screen_res");
+
```
Comment (RunitaiLinden) [inline indra/newview/pipeline.cpp:7169]: Not a blocker for a merge, but should probably just use a static LLCachedControl here for RenderCASSharpness.  "refreshCachedSettings" has a lot of side effects and is very slow, so unless a shader reload or a RenderTarget allocation is required, just use LLCachedControl wherever you want to access the saved setting.

**Review by RunitaiLinden — APPROVED** [review]

File: indra/newview/pipeline.cpp [hunk 2]
```diff
@@ -117,6 +117,17 @@
...
+#define A_GCC 1
+#pragma GCC diagnostic ignored "-Wunused-function"
+#pragma GCC diagnostic ignored "-Wunused-variable"
+#if LL_LINUX
+#pragma GCC diagnostic ignored "-Wrestrict"
+#endif
+#endif
+#define A_CPU 1
+#include "app_settings/shaders/class1/deferred/CASF.glsl" // This is also C++
+
```
Comment (RunitaiLinden) [inline indra/newview/pipeline.cpp:130]: Fine for now, but we should probably break this into its own .cpp file so it doesn't pollute all of pipeline.cpp.

Comment (RyeMutt) [reply indra/newview/pipeline.cpp:130]: Fixed!

Comment (RyeMutt) [reply indra/newview/pipeline.cpp:130]: Yeah I agree on that. Went with the easiest initial integration.

File: indra/newview/pipeline.cpp (hunk 1 above)
Comment (RyeMutt) [reply indra/newview/pipeline.cpp:7167]: Fixed.

**Pull request merged**